tart2ms/tart2ms.py

```diff
@@ -612,27 +674,52 @@ def __twelveball(direction):
             uvw_array = np.array([uu_a, vv_a, ww_a]).T
         elif uvw_generator == 'casacore':
             # need to generate UVW coordinates for zenith positions for the model prediction step
             # if enabled otherwise we can wait till the end (unless we rephase)
-            if phase_center_policy == 'rephase-obs-midpoint' or \
-               phase_center_policy == 'rephase-NCP' or \
-               phase_center_policy == 'rephase-SCP' or \
+            if isinstance(phase_center_policy, SkyCoord) or \
+               phase_center_policy.find('rephase-') == 0 or \
                fill_model:
                 # we must first have accurate uvw coordinates in each different zenith direction
-                if phase_center_policy == 'rephase-obs-midpoint':
+                if phase_center_policy == 'rephase-obs-midpoint' or phase_center_policy == 'no-rephase-obs-midpoint':
                     centroid_direction = zenith_directions[zenith_directions.shape[0]//2, :].reshape(
                         1, 2)
-                elif phase_center_policy == 'rephase-NCP':
-                    centroid_direction = np.array(
-                        [0, np.deg2rad(+90)]).reshape(1, 2)
-                elif phase_center_policy == 'rephase-SCP':
-                    centroid_direction = np.array(
-                        [0, np.deg2rad(-90)]).reshape(1, 2)
+                elif isinstance(phase_center_policy, SkyCoord):
+                    centroid_direction = np.deg2rad(np.array([[phase_center_policy.icrs.ra.value, 
+                                                               phase_center_policy.icrs.dec.value]]).reshape(1, 2))
+                elif phase_center_policy.find("rephase-") == 0:
+                    fn = phase_center_policy.replace("rephase-","")
+                    # 3CRR currently only catalog with special names
+                    catalog_positions = catalog_reader.catalog_factory.from_3CRR(fluxlim15=0.0)
+                    named_positions = read_known_phasings()
+                    fsrc = list(filter(lambda x: x.name == fn, catalog_positions))
+                    if len(fsrc) > 0:
+                        centroid_direction = np.array([[fsrc[0].rarad, fsrc[0].decrad]]).reshape(1, 2)
+                    else: # otherwise it is in the named positions list
+                        fsrc = list(filter(lambda x: x['name'].upper() == fn, named_positions))
+                        if len(fsrc) == 0:
+                            raise RuntimeError(f"Unknown named source {fn}")
+                        if fsrc[0]['position']['FRAME'] == "Special Body":
+                            centroid_direction = np.empty((len(obstime), 2), dtype=float)
+                            for ti, tt in enumerate(obstime):
+                                body = ac.get_body(fn, tt, location=location)
+                                centroid_direction[ti, :] = np.deg2rad(np.array([body.icrs.ra.value, body.icrs.dec.value]))
```

Comment:
Solar system tracking is special in that we compute phases and UVW coordinates for each integration (ie. we go from Zenith to the solar system object because it will have different RA DEC for each integration instead of going from Zenith to a single fringe stopping position. 

The RA DEC as it appears in the database is therefore arbitrary and should be read relative offset instead. This is needed for long exposure images on a solar system object (imagers don't typically combine fields)

Reply:
I don't understand here. Surely this is true regardless of what is being tracked. In each case the object is moving in the telescope frame, and uvw coordinates will need to be recalculated for each integration? This will happen with celestial objects, solar system objects and satellites?